src/Data/DataSource.php

```diff
@@ -193,17 +193,21 @@ public static function resolve_plugins_connection( $source, array $args, AppCont
 	 *
 	 * @throws \Exception
 	 */
-	public static function resolve_post_object( $id, AppContext $context ) {
+	public static function resolve_post_object( $id, AppContext $context, $post_type = 'post' ) {
 
 		if ( empty( $id ) || ! absint( $id ) ) {
 			return null;
 		}
 		$post_id = absint( $id );
-		$context->PostObjectLoader->buffer( [ $post_id ] );
 
-		return new Deferred( function () use ( $post_id, $context ) {
-			return $context->PostObjectLoader->load( $post_id );
-		} );
+		$loader = apply_filters( 'resolve_post_object_loader', 'PostObjectLoader', $post_id, $context, $post_type );
```

Comment:
Hey, so I was thinking the filter should happen in the Context, not here, because the Loaders can be used outside of these DataSource methods

Any resolve function should be able to do something like `$context->loaders['post_object']->buffer([$post_id]);`

So, instead of having an action in context, we would have one filter, and `AppContext->loaders` would be an array of loaders, passed through the filter. So folks could add custom loaders, or even replace loaders using that filter. 

Then the loaders are centralized and accessible to all resolvers.

Reply:
@kidunot89 mentioning you to make sure you get notified of my comment 😄